eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/SegmentNr.scala

```diff
@@ -3,18 +3,90 @@ package com.evolutiongaming.kafka.journal.eventual.cassandra
 import cats.kernel.Eq
 import cats.syntax.all.*
 import cats.{Applicative, Id, Monad, Order, Show}
-import com.evolutiongaming.kafka.journal.SeqNr
 import com.evolutiongaming.kafka.journal.util.Fail
 import com.evolutiongaming.kafka.journal.util.Fail.implicits.*
+import com.evolutiongaming.kafka.journal.{Key, SeqNr}
 import com.evolutiongaming.scassandra.{DecodeByName, DecodeRow, EncodeByName, EncodeRow}
 
+/**
+  * Segment fields in `journal` and `metajournal` tables. Part of clustering key in both tables.
+  * 
+  * [[SegmentNr]] can be limited: in `metajournal` table cannot be more than [[Segments]] thus:
+  * ```
+  *    val metaJournalSegmentNr: SegmentNr = ???
+  *    val metaJournalSegments: Segments = ???
+  *    metaJournalSegmentNr <= metaJournalSegments // have to be true
+  * ```
+  * 
+  * In `journal` table [[SegmentNr]] is not limited but rather each segment contains not more than [[SegmentSize]] events:
+  * ```
+  *   val journalSegmentNr: SegmentNr = ???
+  *   val journalSegmentSize: SegmentSize = ???
+  *   journalSegmentNr.value > journalSegmentSize.value // can be true but not have to be
+  * ```
+  * 
+  * Used to partition the `journal` table into segments thus distribute data between Cassandra partitions.
+  * For `journal` table [[SegmentNr]] calculated based on [[SeqNr]] and [[SegmentSize]] in [[SegmentNr.journal]]:
+  * ```
+  *    val seqNr: SeqNr             = SeqNr.unsafe(45629)
+  *    val segmentSize: SegmentSize = SegmentSize.default // 10000
+  *    val segmentNr: SegmentNr     = SegmentNr.journal(seqNr, segmentSize) // (45629 - 1) / 10000 = 4 where `1` is SeqNr.min
+  * ```  
+  * 
+  * Used to partition the `metajournal` table into segments thus simplify iterating over all records in the table.
+  * For `metajournal` table [[SegmentNr]] calculated based on [[Key]] and [[Segments]] in [[SegmentNr.metaJournal]]:
+  * ```
+  *    val key: Key             = Key("journal ID", "journal topic") // key.id.toLowerCase.hashCode = 969045668
+  *    val segments: Segments   = Segments.default // 10000
+  *    val segmentNr: SegmentNr = SegmentNr.metaJournal(key, segments) // Math.abs("journal ID".toLowerCase.hashCode) % 10000 = 5668
+  * ``` 
+  * 
+  * @see [[Segments]] for `metajournal` table
+  * @see [[SegmentNrs]] for `metajournal` table
+  * @see [[SegmentOf]] for `metajournal` table
+  * @see [[SegmentNrsOf]] for `metajournal` table
+  * 
+  * @see [[SegmentSize]] for `journal` table
+  * @see [[Segment]] for `journal` table
```

Comment:
I'd like to rename them to not have similar names for different tables, but I cannot provide any suggestion, yet

Reply:
I see no big issue in using same case class for almost same values in different tables